charts/newrelic-infrastructure-v3/templates/controlplane/daemonset.yaml

```diff
@@ -11,9 +11,15 @@ metadata:
   annotations: {{ . | nindent 4 }}
   {{- end }}
 spec:
-  {{- if eq .Values.controlPlane.kind "DaemonSet"}}
+  {{- if eq .Values.controlPlane.kind "DaemonSet" }}
   {{- with .Values.updateStrategy }}
-  updateStrategy: {{ toYaml . | nindent 4 }}
+  updateStrategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- else if eq .Values.controlPlane.kind "Deployment" }}
+  {{- with .Values.controlPlane.strategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
```

Comment:
I don't think there's a problem with the `hostNetwork` helper. It does what it needs to do: Checking if the user _allows_ (not sets) running with `hostNetwork` or not.

I think the issue here is that `hostNetwork` is just not compatible with deploying a Deployment (if you open a port as we do), so my approach is to change the strategy and _not_ hostNetwork. This way the user has full control over both the strategy and hostNetwork and can tweak them however they please, while keeping a default that works.

Why should we actively prevent the user from running `hostNetwork`? It might not be a common use case, but my motto here is to avoid as much as possible for the user to need to fork the chart, open issues, or the worst of all, deploy manifests modified by hand.

Reply:
So what you want to do is the same as in `lowDataMode`: https://github.com/newrelic/helm-charts/pull/632/files#diff-ad90df5b5bd9a1c31f58e9ee28b3ad8ac425e6d720533d960f1495120e98cdb8R122-R139

Delete `unprivilegedHostNetwork` and `hostNetwork` from the chart's `values.yaml` (comment them) and change the default value depending on the `Kind` of the control plane. If it is `DaemonSet`, `hostNetwork` is `true`, else is `false`, unless the user have put a value in its `values.yaml` that overrides any default.

Could this be a better approach? Would you like that I do that logic in my PR for comparing it?

Reply:
> Delete unprivilegedHostNetwork and hostNetwork from the chart's values.yaml (comment them) and change the default value depending on the Kind of the control plane.

On the contrary, I suggest to keep all of them and let users combine them however they want.

Reply:
Delete as in "comment them", not as in "delete its functionality".

`lowDataMode` has it commented to make it overridable: https://github.com/newrelic/helm-charts/pull/632/files#diff-b5aab5e71b9779d8667e1a61c2f7ac342c2833e4c74ccafebdc77b8d43792be1R26